staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go

```diff
@@ -126,19 +126,25 @@ func BeforeDelete(strategy RESTDeleteStrategy, ctx context.Context, obj runtime.
 		return false, true, nil
 	}
 
+	// `CheckGracefulDelete` will be implemented by specific strategy
 	if !gracefulStrategy.CheckGracefulDelete(ctx, obj, options) {
 		return false, false, nil
 	}
-	now := metav1.NewTime(metav1.Now().Add(time.Second * time.Duration(*options.GracePeriodSeconds)))
-	objectMeta.SetDeletionTimestamp(&now)
-	objectMeta.SetDeletionGracePeriodSeconds(options.GracePeriodSeconds)
-	// If it's the first graceful deletion we are going to set the DeletionTimestamp to non-nil.
-	// Controllers of the object that's being deleted shouldn't take any nontrivial actions, hence its behavior changes.
-	// Thus we need to bump object's Generation (if set). This handles generation bump during graceful deletion.
-	// The bump for objects that don't support graceful deletion is handled in pkg/registry/generic/registry/store.go.
-	if objectMeta.GetGeneration() > 0 {
-		objectMeta.SetGeneration(objectMeta.GetGeneration() + 1)
+
```

Comment:
if CheckGracefulDelete returns true, options.GracePeriodSeconds must be non-nil (that's what tells the deletion how "graceful" to be).

A nil check could be ok, but it's an internal error if options.GracePeriodSeconds is nil at this point, and the error should fail the request with an internal error. The docs on the RESTGracefulDeleteStrategy method should be updated as well.

Reply:
I expected something more like this inserted at line 135, without any other changes:
```go
if options.GracePeriodSeconds == nil {
  return false, false, ... internal error ...
}
```